core/src/main/scala/kafka/cluster/Replica.scala

```diff
@@ -42,6 +42,10 @@ class Replica(val brokerId: Int, val topicPartition: TopicPartition) extends Log
   // the LEO of leader at time t. This is used to determine the lag of this follower and ISR of this partition.
   @volatile private[this] var _lastCaughtUpTimeMs = 0L
 
+  // highWatermark is the leader's high watermark after the most recent FetchRequest from this follower. This is
+  // used to determine the maximum HW this follower knows about
+  @volatile private[this] var _highWatermark = 0L
```

Comment:
Maybe a name like `lastSentHighWatermark` would make it clearer that we do not know this value has actually been acked by the follower. Similar for the getter. This just reduces the risk of making invalid assumptions if we use this in other contexts.